### DISCUSSION TITLE
Is it possible to set the rules property dynamically?

### DISCUSSION THREAD

**Discussion by DanielGSoftware** [livewire/livewire#1839] [1. Help] [answered]
I can't set the value for the rules properly directly, because I use a function to retrieve the rules. If I try this, I get the `php expression is not allowed as field default value` error. When I'm trying to set the rules via the mount() method, I get the `Livewire encountered corrupt data when trying to hydrate the [component-name] component.` error.

See my code below:

```
public array $rules = [];

    public function mount(LoanApplication $loanApplication): void
    {
        $this->financialSituation = $loanApplication->financial_situation;
        $this->rules = $this->financialSituation->getRules();
    }
```

Does someone know how to set the rules property? I've been stuck with this for a while now, so thanks in advance for the help, I really appreciate it

**Comment by DanielGSoftware** [comment, marked as answer]
I found the problem on why I was getting the `Livewire encountered corrupt data when trying to hydrate the [component-name] component.` error. When I'm calling `$this->financialSituation->getRules()`, there are extra fields in that array that I don't use in my form. Removing those extra validation fields fixes the error.

**Comment by dominiced** [comment]
You can also replace the rules property with a protected rules() method, and return back the array when it's called, and that's a good way to use dynamic properties. Livewire will automatically pick up the method if the property isn't used.
